gcovr/utils.py

```diff
@@ -37,6 +37,32 @@ def already_visited(self, path):
         return False
 
 
+if sys.platform == "win32":
+    if sys.version_info >= (3, 8):
+        gcov_realpath = os.path.realpath
+    else:
+        # Only used for old python versions. Function can be treated as stable.
+        from nt import _getfinalpathname
+
+        DOS_DEVICE_PATH_PREFIX = "\\\\?\\"
+        DOS_DEVICE_PATH_PREFIX_UNC = DOS_DEVICE_PATH_PREFIX + "UNC\\"
+
+        def gcov_realpath(path):
```

Comment:
I'd probably have written this function with multiple returns, but that's a matter of personal taste:

```python
def gcovr_realpath(path: str) -> str:
    realpath = os.path.realpath(path)

    # only resolve symlinks if the path exists
    if not os.path.exists(realpath):
        return realpath

    resolved_path = _getfinalpathname(realpath)

    # only strip DOS prefix from resolved path if necessary to match realpath
    if realpath.startswith(DOS_DEVICE_PATH_PREFIX):
        return resolved_path

    if resolved_path.startswith(DOS_DEVICE_PATH_PREFIX_UNC):
        return resolved_path[len(DOS_DEVICE_PATH_PREFIX_UNC):]
    if resolved_path.startswith(DOS_DEVICE_PATH_PREFIX):
        return resolved_path[len(DOS_DEVICE_PATH_PREFIX):]
    return resolved_path
```

In any case, does it make sense to name the function GCOV-realpath? This does not seem to relate to the GCC tool. Calling it GCOV**R**-realpath could make sense, but that prefix seems redundant within our codebase.

Reply:
I come from C development in the automotive and we use the Misra coding rules. There is one rule for usage of a single return. :-)